src/node_wasm_streaming.cc

```diff
@@ -0,0 +1,377 @@
+// resolved value of the Promise passed to .compileStreaming (aka
+// "Response") and a WasmStreaming object.  The callback should submit
+// WASM data by calling .onBytesReceived on the WASM streaming object
+// followed by .Finish.
+void SetCallback(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  if (args.Length() == 0 || !args[0]->IsFunction())
+    env->ThrowError("First argument must be a function");
+  else
+    env->set_wasm_streaming_callback(args[0].As<Function>());
+}
+
+void Initialize(Local<Object> target,
+                Local<Value> unused,
+                Local<Context> context,
+                void* priv) {
+  Environment* env = Environment::GetCurrent(context);
+  Isolate* isolate = env->isolate();
+
+  env->SetMethod(target, "setCallback", SetCallback);
+
+  Local<FunctionTemplate> wasm_streaming = FunctionTemplate::New(isolate);
+  auto classname = FIXED_ONE_BYTE_STRING(env->isolate(), "WasmStreaming");
+  wasm_streaming->SetClassName(classname);
+
+  env->SetProtoMethod(wasm_streaming, "onBytesReceived",
+                      WasmStreaming::OnBytesReceived);
+  env->SetProtoMethod(wasm_streaming, "finish",
+                      WasmStreaming::Finish);
+  env->SetProtoMethod(wasm_streaming, "abort",
+                      WasmStreaming::Abort);
+  env->SetProtoMethod(wasm_streaming, "setCompiledModuleBytes",
+                      WasmStreaming::SetCompiledModuleBytes);
+  env->SetProtoMethod(wasm_streaming, "setUrl",
+                      WasmStreaming::SetUrl);
+
+  Local<ObjectTemplate> wasm_streaming_tmpl =
+    wasm_streaming->InstanceTemplate();
+  wasm_streaming_tmpl->SetInternalFieldCount(1);
+
+  DCHECK(env->wasm_streaming_instance_template().IsEmpty());
+  env->set_wasm_streaming_instance_template(wasm_streaming_tmpl);
+}
+
+}  // anonymous namespace
+
+// Implements streaming for WebAssembly.compileStreaming (V8 callback)
+// by forwarding the call to a user-registered JS function.
+void WasmStreamingCallback(const FunctionCallbackInfo<Value>& args) {
+  Isolate* isolate = args.GetIsolate();
+  Environment* env = Environment::GetCurrent(isolate);
+  CHECK(env);
+  auto wasm_streaming = v8::WasmStreaming::Unpack(isolate, args.Data());
+  DCHECK(wasm_streaming);
+  Local<Function> cb = env->wasm_streaming_callback();
+  TryCatchScope try_catch(env);
+  if (cb.IsEmpty()) {
+    env->ThrowError("WASM streaming not ready");
+  } else {
+    auto tmpl = env->wasm_streaming_instance_template();
+    DCHECK(!tmpl.IsEmpty());
+    Local<Object> wasm_streaming_object;
+    if (tmpl->NewInstance(env->context()).ToLocal(&wasm_streaming_object)) {
+      new WasmStreaming(env, wasm_streaming_object, wasm_streaming);
+      // cb(Response, WasmStreaming)
+      Local<Value> cb_args[] = {args[0], wasm_streaming_object};
+      USE(cb->Call(env->context(), Undefined(isolate),
+                   arraysize(cb_args), cb_args));
+    }
+  }
+  if (try_catch.HasCaught())
```

Comment:
I think you may also need to check 
```suggestion
  if (try_catch.HasCaught() && !try_catch.HasTerminated())
```

(if the exception is a termination exception, `try_catch.Exception()` is not going to give you a meaningful value)